airbyte-integrations/connectors/destination-databricks/src/main/kotlin/io/airbyte/integrations/destination/databricks/jdbc/DatabricksDestinationHandler.kt

```diff
@@ -0,0 +1,278 @@
+import io.airbyte.cdk.integrations.destination.jdbc.TableDefinition
+import io.airbyte.integrations.base.destination.typing_deduping.AirbyteProtocolType.STRING
+import io.airbyte.integrations.base.destination.typing_deduping.AirbyteProtocolType.TIMESTAMP_WITH_TIMEZONE
+import io.airbyte.integrations.base.destination.typing_deduping.DestinationHandler
+import io.airbyte.integrations.base.destination.typing_deduping.DestinationInitialStatus
+import io.airbyte.integrations.base.destination.typing_deduping.InitialRawTableStatus
+import io.airbyte.integrations.base.destination.typing_deduping.Sql
+import io.airbyte.integrations.base.destination.typing_deduping.StreamConfig
+import io.airbyte.integrations.base.destination.typing_deduping.StreamId
+import io.airbyte.integrations.base.destination.typing_deduping.migrators.MinimumDestinationState
+import io.airbyte.protocol.models.v0.DestinationSyncMode
+import io.github.oshai.kotlinlogging.KotlinLogging
+import java.sql.Connection
+import java.sql.ResultSet
+import java.sql.SQLException
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+import java.util.*
+import kotlin.streams.asSequence
+
+class DatabricksDestinationHandler(
+    private val databaseName: String,
+    private val jdbcDatabase: JdbcDatabase,
+) : DestinationHandler<MinimumDestinationState.Impl> {
+
+    private val log = KotlinLogging.logger {}
+    private val abRawId = DatabricksSqlGenerator.AB_RAW_ID
+    private val abExtractedAt = DatabricksSqlGenerator.AB_EXTRACTED_AT
+    private val abMeta = DatabricksSqlGenerator.AB_META
+
+    override fun execute(sql: Sql) {
+        val transactions: List<List<String>> = sql.transactions
+        val queryId = UUID.randomUUID()
+        for (transaction in transactions) {
+            val transactionId = UUID.randomUUID()
+            log.info { "Executing sql $queryId-$transactionId: ${transactions.joinToString("\n")}" }
+            val startTime = System.currentTimeMillis()
+
+            try {
+                // Databricks DOES NOT support autocommit false. ACID guarantees are within a single
+                // table
+                // so only MERGE is the supported way if updates/deletes to be done in source &
+                // target table.
+                // CREATE OR REPLACE...SELECT * from... for swapping a table.
+                transaction.forEach { jdbcDatabase.execute(it) }
+            } catch (e: SQLException) {
+                log.error(e) {
+                    "Sql $queryId-$transactionId failed in ${System.currentTimeMillis() - startTime} ms"
+                }
+                throw e
+            }
+            log.info {
+                "Sql $queryId-$transactionId completed in ${System.currentTimeMillis() - startTime} ms"
+            }
+        }
+    }
+
+    override fun gatherInitialState(
+        streamConfigs: List<StreamConfig>
+    ): List<DestinationInitialStatus<MinimumDestinationState.Impl>> {
```

Comment:
curious what you think the right approach for the temp raw table stuff will be (for refreshes). I think there's a few different cases:
* merge refresh / normal sync: we need to know whether the temp raw table exists and whether the real raw table exists _before_ fetching the "real" initial state. (this is so that we can copy the temp raw records into the real raw table)
* truncate refresh: we need to know whether the temp raw table exists (boolean) and also `select generation_id limit 1` from it. Then we can just assume `minTimestamp = Optional.empty()`.

Is that maybe a separate `gatherTemporaryRawTableState` step, which happens prior to `gatherInitialState`?

Reply:
`DestinationInitialStatus` can be loosely defined and evolved probably and have as much info from either platform's injection of information or us doing gathering from existing warehouse state.